posydon/binary_evol/singlestar.py

```diff
@@ -177,24 +177,35 @@ def append_state(self):
         for item in STARPROPERTIES:
             getattr(self, item + '_history').append(getattr(self, item))
 
-    def restore(self, i=0, delete_history=True):
-        """Restore the object to the i-th state.
+    def restore(self, i=0, hooks=[]):
```

Comment:
FYI setting mutables (e.g. an empty list) in kwargs may lead to unexpected behavior if any updates occur within the function. Kwargs are bound during _define time_ and not _run time_. Catching it like this guarantees to avoid this issue:
```python
def restore(self, i=0, hooks=None):
    ...
    if hooks is None:
        hooks = []
```